README.md

```diff
@@ -1,2 +1,100 @@
+
+### Response
+The HAL document always includes the following attributes
+
+and, when applicable, the following links
+- **self**: a parameterized link to the requested collection page
+- **next**: the link to the next page of resources in the collection, if any, keeping the same option for size and sorting 
+- **previous**: the link to the previous page of resources in the collection, if any, keeping the same option for size and sorting
+- **first**: the link to the first page of resources in the collection, keeping the same option for size and sorting
+- **last**: the link to the last page of resources in the collection, keeping the same option for size and sorting
+
+### Out of bound pages
+In the case that the request parameters lead to a page outside the result set an empty page should be returned with the links needed to go to the first and last page of the result set if the results set is not empty and the total number of resources in the collection
+
+### Error Code
+400 Bad Request. If an unknown sort criteria is requested or a not valid ordering keyword is specified
+
+## ETags & conditional headers
+The ETag header (<http://tools.ietf.org/html/rfc7232#section-2.3>) provides a way to tag resources. This can prevent clients from overriding each other while also making it possible to reduce unnecessary calls. It is expected that all the returned document have an ETag
+
+The ETag value can be used with in GET request with the *If-None-Match* conditional header. If the header MATCHES the ETag, the API will conclude nothing has changed, and instead of sending a copy of the resource, an HTTP 304 Not Modified status code is returned.
+
+The ETag value can be also used with DELETE, POST, PUT and PATCH with the *If-Match* conditional header to avoid to perform action on changed resources (concurrency issues, optimistic lock approach).
+
+Finally, when possible the If-Modified-Since header in GET request should be respected. If the resource has been not modified since the value of the Header the API should return an
+HTTP 304 Not Modified status code. Resources that support the *If-Modified-Since* header *MUST* return the Last-Modified Header in the GET response, such header *MUST BE NOT* returned by resources not able to manage the If-Modified-Since header.  
+
+## ALPS - Application Level Profile Semantics
+It is expected support for the ALPS metadata (<http://alps.io/>), so a profile link MUST exists from the root of API.
+A profile link as defined in RFC 6906 (<https://tools.ietf.org/html/rfc6906>), is a place to include application level details. The ALPS draft spec (http://tools.ietf.org/html/draft-amundsen-richardson-foster-alps-00)
+
+## Endpoints
+At the root of the api a HAL document MUST list all the primary endpoints allowing full discovery of the current version of the API.
+
+[Documentation of the defined endpoints](endpoints.md) 
```

Comment:
It would be useful to note a recommended mechanism for creating (1)a plugin that is intended for redistribution and (2)a local customization that an instance might develop.